### ISSUE TITLE
Mermaid Tooltips are not working

### ISSUE THREAD

**Issue by joskfg** [squidfunk/mkdocs-material#4686]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Mermaid.js is not rendering the Tooltips

### Expected behaviour

I'd expect to see the tooltip when I put the cursor over the node with tooltip.

### Actual behaviour

The tooltip is now shown when you put the cursor over the node with tooltip.

I think it is because it uses this code for render:
```
mermaid.mermaidAPI.render(id, el.textContent, (svg: string) => {

      /* Create a shadow root and inject diagram */
      const shadow = host.attachShadow({ mode: "closed" })
      shadow.innerHTML = svg

      /* Replace code block with diagram */
      el.replaceWith(host)
    })
```

So it just recover the SVG data and not the div that is needed to show the tooltip.

### Steps to reproduce

Create a simple flowchart diagram like
```
```mermaid
flowchart LR
    TEST
    click TEST callback "I should be able to see this"
```
```

### Package versions

Using image: squidfunk/mkdocs-material:8.5.8

### Configuration

```yaml
site_name: 'my docs'
site_url: 'http://127.0.0.1'
# Extensions
markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
```

### System information

- Operating system: any
- Browser: any

**Comment by facelessuser** [comment]
I believe this has been mentioned before, but IIRC this is due to the fact that Mermaid has to be rendered within a shadow DOM due to how they handle some of their IDs and such (I don't have the issue number off the top of my head, but I believe the Mermaid issue is still open). I know that Mermaid has been making some progress in this area, but I have not evaluated if Mermaid support can yet properly function without side-effects if dropping the shadow DOM.

**Comment by squidfunk** [comment]
Thanks for reporting. I'm not sure I understand the problem. [If I run your example in the Mermaid Live editor](https://mermaid.live/edit#pako:eNo1TrsOwjAM_BXLc78gMwxIsNCOXtzEkKhJg1JHCFX9d0IrbrmHTqdb0WYnaPAR89t6LgrXO83QMJz74VA2BjvtHizHOHJzhBdYfK7RwSjAYxTQDIs08mEhxA6TlMTBtfH1t0OoXpIQmiYdl4mQ5q31uGruP7NFo6VKh_XlWOUU-Fk4_UNxQXO5HWf3z9sXWKc94w), I don't see any tooltip being displayed. Can you link to a fully working example and explain what we have to do to see the tooltip?

**Comment by joskfg** [comment]
You can see the [example](https://jsfiddle.net/s37cjoau/3/) from the [documentation page](https://mermaid-js.github.io/mermaid/#/flowchart?id=interaction).

As far as I saw mermaid create a div after the SVG with the class **.mermaidTooltip** so it is shown when your mouse is over the specific node. One possible approach is to check the SVG content searching that tooltips and create the divs manually.

**Comment by squidfunk** [comment]
Thanks for the input. So I partly got this working, we just needed to execute the bind handlers if there are any. I added basic support in 3c326e80d. However, the positioning of the tooltip is wrong, as it's appended to the end of the body. I have no idea why, but it seems as is this is a problem with Mermaid itself:

<img width="840" alt="Bildschirmfoto 2022-11-30 um 22 08 53" src="https://user-images.githubusercontent.com/932156/204910049-3a8d6f8e-5463-44be-99c0-2c8c7349a7ae.png">

I've dug around and found https://github.com/mermaid-js/mermaid/issues/3111, which seems related. I invite you to open an issue on the Mermaid issue tracker to ask for support, but I guess there's nothing we can do on our side to make this work correctly.

PR appreciated.
